river/tree/_objective/base_objective.py

```diff
@@ -0,0 +1,32 @@
+from abc import ABCMeta
+
+from .._utils import GradHess
+
+
+class BaseObjective(metaclass=ABCMeta):
+    """ Base class to implement optimization objectives used in Streaming Gradient Trees. """
+
+    def compute_derivatives(self, y: float, y_pred: float) -> GradHess:
```

Comment:
Suggestion for clarity: `y_true`, `y_pred`

Reply:
Cool! Refactored in commit [940122d](https://github.com/online-ml/river/pull/458/commits/940122dfb81b6e62d7f43a2e51cfb113b90d899d).